runtime/jvmti/jvmtiHelpers.c

```diff
@@ -700,10 +700,9 @@ javaOffloadSwitchOn(J9VMThread * currentThread, UDATA eventNumber, UDATA javaOff
 
 
 jint
-getThreadState(J9VMThread *currentThread, j9object_t threadObject)
+getThreadStateHelper(J9VMThread *currentThread, j9object_t threadObject, J9VMThread *vmThread)
```

Comment:
Why break this into it's own function rather than just adding the new code at the end?

Reply:
Now, `getThreadState` is strictly used with platform (non-virtual) threads. `getVirtualThreadState` used to rely on `getThreadState`, which has been changed to the helper. `getVirtualThreadState` malfunctions if the new code, which is platform thread specific, is added at the end.